src/ImageSharp/Processing/Processors/Normalization/AdaptiveHistogramEqualizationProcessor{TPixel}.cs

```diff
@@ -614,7 +619,9 @@ public void Invoke(in RowInterval rows)
                                 for (int dx = x; dx < xlimit; dx++)
                                 {
                                     int luminance = GetLuminance(rowSpan[dx], this.luminanceLevels);
-                                    histogram[luminance]++;
+
+                                    // This is safe. The index maxes out to the span length.
+                                    Unsafe.Add(ref histogramBase, luminance)++;
```

Comment:
Feel free to ignore, but I would prefer to stick with span indexers systematically in less critical processors. Span indexer is quite fast these days and `GetLuminance` is way more expensive than the indexer.

Reply:
Will revert. Should be replaced anyway with a span based version.